src/Connector.AzureDataLake/Connector/AzureDataLakeConnector.cs

```diff
@@ -135,9 +350,86 @@ public override Task<IAsyncEnumerable<ConnectorLatestEntityPersistInfo>> GetLate
+                await _client.EnsureDataLakeDirectoryExist(jobData);
+
+                if (jobData.IsStreamCacheEnabled)
+                {
+                    if (string.IsNullOrWhiteSpace(jobData.StreamCacheConnectionString))
+                    {
+                        return new ConnectionVerificationResult(false, $"Stream cache connection string must be valid when buffer is enabled.");
+                    }
+                    await using var connection = new SqlConnection(jobData.StreamCacheConnectionString);
+                    await connection.OpenAsync();
+
+                    await VerifyTableOperations(connection);
+                }
+                else
+                {
+                    if (!AzureDataLakeConstants.OutputFormats.Json.Equals(jobData.OutputFormat, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return new ConnectionVerificationResult(false, $"Only JSON is supported when stream cache is disabled.");
+                    }
+                }
+                return new ConnectionVerificationResult(true);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Error verifying connection");
+                return new ConnectionVerificationResult(false, e.Message);
+            }
+        }
+
+        private async Task VerifyTableOperations(SqlConnection connection)
+        {
+            var testStreamId = Guid.NewGuid();
+            var testTableName = GetCacheTableName(testStreamId, true);
+
+            var entityId = Guid.NewGuid();
+            var data = new Dictionary<string, object>
+            {
+                [AzureDataLakeConstants.IdKey] = entityId,
+                ["testColumn"] = 1234,
+            };
+
+            var dataValueTypes = new Dictionary<string, Type>
+            {
+                [AzureDataLakeConstants.IdKey] = typeof(Guid),
+                ["testColumn"] = typeof(string),
+            };
 
-            return new ConnectionVerificationResult(true);
+            await EnsureCacheTableExists(
+                    connection,
+                    testTableName,
+                    data.Keys.Except(new[] { AzureDataLakeConstants.IdKey }).ToList(),
+                    dataValueTypes);
+            try
+            {
+                var baseSyncItem = new SyncItem(testStreamId, entityId, VersionChangeType.Added, data, dataValueTypes);
+                await VerifyOperation(connection, testTableName, baseSyncItem);
+                await VerifyOperation(connection, testTableName, baseSyncItem with { ChangeType = VersionChangeType.Changed });
+                await VerifyOperation(connection, testTableName, baseSyncItem with { ChangeType = VersionChangeType.Removed });
+            }
+            finally
+            {
+                await DeleteCacheTableIfExists(connection, testTableName);
```

Comment:
see previous comment about using a transaction so we don't need to delete the table

Reply:
as discussed, we put this inside transactions so that we can test delete too